src/loaddb/load_semantic_helper.cpp

```diff
@@ -126,49 +136,30 @@ namespace cubload
   string_type *
   semantic_helper::make_string_by_buffer ()
   {
-    string_type *str = make_string ();
-    if (str == NULL)
-      {
-	return NULL;
-      }
-
-    str->size = m_qstr_buf_idx - 1;
+    string_type *str = NULL;
+    std::size_t str_size = m_qstr_buf_idx - 1;
 
-    if (!m_use_qstr_buffer)
+    if (!m_use_qstr_buf)
       {
-	str->val = m_qstr_buf_p;
-	str->need_free_val = false;
+	str = make_string (m_qstr_buf_p, str_size, false);
       }
     else
       {
-	if (use_copy_buf_pool (str->size))
+	if (use_copy_buf_pool (str_size))
 	  {
-	    str->val = &m_copy_buf_pool[m_copy_buf_pool_idx++][0];
-	    str->need_free_val = false;
+	    str = make_string (&m_copy_buf_pool[m_copy_buf_pool_idx++][0], str_size, false);
 	  }
 	else
 	  {
-	    str->val = (char *) malloc (m_qstr_buf_idx);
-	    if (str->val == NULL)
-	      {
-		er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_LDR_MEMORY_ERROR, 0);
-	      }
-
-	    str->need_free_val = true;
+	    str = make_string (new char[m_qstr_buf_idx], str_size, true);
 	  }
       }
 
-    if (str->val == NULL)
-      {
-	free_string (&str);
-	return NULL;
-      }
-
-    memcpy (str->val, m_qstr_buf_p, m_qstr_buf_idx);
+    std::memcpy (str->val, m_qstr_buf_p, m_qstr_buf_idx);
```

Comment:
`std::memcpy (str->val, m_qstr_buf_p, m_qstr_buf_idx);` is not necessary in `!m_use_qstr_buf` case.